From 811c2a862894b1aacc5f15ea7fda6a1dcf822348 Mon Sep 17 00:00:00 2001
From: Marat Shamshutdinov <marat@citrus-soft.ru>
Date: Thu, 5 May 2022 10:50:12 +0300
Subject: [PATCH 2/2] Hotfix https://github.com/DHFinance/dhf-pay-php/pull/4

---
 src/DHFPay.php | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/DHFPay.php b/src/DHFPay.php
index ef9f2c5..d71f718 100644
--- a/src/DHFPay.php
+++ b/src/DHFPay.php
@@ -77,7 +77,7 @@ class DHFPay
     {
         try {

-            $newresponse = $this->client->request(
+            $newresponse = (string)$this->client->request(
                 $method,
                 $uri,
                 [
@@ -88,7 +88,7 @@ class DHFPay
                         ],
                     RequestOptions::JSON => $body
                 ]
-            )->getBody()->getContents();
+            )->getBody();

         } catch (RequestException $exception) {
             throw $this->processException($exception);
--
2.20.1

